src/site/antora/modules/ROOT/pages/manual/cloud.adoc

```diff
@@ -15,549 +15,267 @@
 
-The JsonTemplateLayout can also be used to generate JSON that matches the GELF specification which can format
-the message attribute using a pattern in accordance with the PatternLayout. For example, the following
-template, named EnhancedGelf.json, can be used to generate GELF-compliant data that can be passed to Logstash.
-With this template the message attribute will include the thread id, level, specific ThreadContext attributes,
-the class name, method name, and line number as well as the message. If an exception is included it will also
-be included with newlines. This format follows very closely what you would see in a typical log file on disk
-using the PatternLayout but has the additional advantage of including the attributes as separate fields that
-can be queried.
+In short, *we don't recommend writing logs to files*.
 
-[source,json]
-----
-{
-    "version": "1.1",
-    "host": "${hostName}",
-    "short_message": {
-        "$resolver": "message",
-        "stringified": true
-    },
-    "full_message": {
-        "$resolver": "message",
-        "pattern": "[%t] %-5p %X{requestId, sessionId, loginId, userId, ipAddress, corpAcctNumber} %C{1.}.%M:%L - %m",
-        "stringified": true
-    },
-    "timestamp": {
-        "$resolver": "timestamp",
-        "epoch": {
-            "unit": "secs"
-        }
-    },
-    "level": {
-        "$resolver": "level",
-        "field": "severity",
-        "severity": {
-            "field": "code"
-        }
-    },
-    "_logger": {
-        "$resolver": "logger",
-        "field": "name"
-    },
-    "_thread": {
-        "$resolver": "thread",
-        "field": "name"
-    },
-    "_mdc": {
-        "$resolver": "mdc",
-        "flatten": {
-            "prefix": "_"
-        },
-        "stringified": true
-    }
-}
-----
+A common excuse for writing logs to files is to use them as a buffer prior to ingesting them either to the logging system proxy or to the log storage system.
+Even in this case, most of the time it is sign of bad architecture design, and costs extra resources and efficiency degradation.
 
-The logging configuration to use this template would be
+[#separate-config]
+=== Separate logging configuration from the application
 
-[source,xml]
+We strongly advise you to separate the logging configuration from the application and couple them in an environment-specific way.
+This will allow you to
+
+* Address environment-specific configurations (e.g., logging verbosity needs of test and production can be different)
+* Ensure Log4j configuration changes applies to all affected Log4j-using software without the need to manually update their Log4j configuration one by one
+
+How to implement this separation pretty much depends on your setup.
+We will share some recommended approaches to give you an idea:
+
+Ship configuration files along with the application, and choose a subset of them during deployment::
```

Comment:
There is literally half a dozen ways to deploy the configuration.

1. Providing `log4j2.configurationFile` that points to a centralized configuration store,
2. Merging different configuration files (why not merge a local "fail-over" config with a remote one?),
3. Use arbiters to select the logging profile,
4. Use property sources to configure the logging level,
5. Mounting an external volume on the container,
6. Custom configurations.

I think we should point the user to the appropriate chapter and let them choose.

Reply:
Improved the text in e03faec93814ea5f57e5c8fdc3a2802801473cbf.

> * Use arbiters to select the logging profile
> * Custom configurations

This section is about _recommended_ approaches, I don't think these would qualify so. No modern SaaS, cloud, service mesh, etc. deployment works with arbiters-minded configuration – i.e., one configuration for all environments.

> Use property sources to configure the logging level

I don't understand how this is relevant.

> Mounting an external volume on the container,

I already explained this below.

While reviewing this section please keep in mind that we share recommended best practices, not every single possible way of deploying a Log4j configuration. Judging from my experience, *all* companies I know of use one of the two techniques I cited here.

Reply:
> This section is about _recommended_ approaches, I don't think these would qualify so. No modern SaaS, cloud, service mesh, etc. deployment works with arbiters-minded configuration – i.e., one configuration for all environments.

I wouldn't be sure that arbiters (especially `SpringBootArbiter`) are not used in modern SaaS.
If they use Spring Boot profiles, they either use `SpringBootArbiter` or separate configurations.
I am not really sure, which one is more popular or which one should be recommended.

> While reviewing this section please keep in mind that we share recommended best practices, not every single possible way of deploying a Log4j configuration. Judging from my experience, _all_ companies I know of use one of the two techniques I cited here.

I rely on your experience here.
Most companies I know are SMEs and are not using cloud environments.